### ISSUE TITLE
iree-compile fails with 'failed to distribute' on single-batch layer norm for gfx942

### ISSUE THREAD

**Issue by sjain-stanford** [iree-org/iree#23650]
## Description

`iree-compile` fails with `'func.func' op failed to distribute` when compiling a `torch.aten.native_layer_norm` operation with a single-batch input shape `[1,128,64,32]` targeting ROCm `gfx942` (MI300X/MI325X). The same layer norm compiles successfully with a multi-batch input shape such as `[16,128,64,32]`.

## Steps to reproduce

Minimal MLIR repro and compile command: https://gist.github.com/sjain-stanford/ed40683ce3b67d0bb3b5dda6db99cd90

```
iree-compile iree-compile-input.mlir \
  --iree-hal-target-backends=rocm \
  --iree-rocm-target=mi300x \
  --iree-opt-level=O3 \
  --iree-preprocessing-pass-pipeline='builtin.module(util.func(iree-preprocessing-convert-conv-filter-to-channels-last))' \
  --iree-flow-enable-pad-handling \
  --iree-global-opt-propagate-transposes-through-conv \
  --iree-global-opt-enable-sink-transpose-through-pad \
  --iree-dispatch-creation-enable-fuse-padding-into-linalg-consumer-ops \
  --iree-dispatch-creation-enable-aggressive-reshape-movement \
  --iree-dispatch-creation-enable-split-reduction \
  --iree-torch-externalize-transients \
  -o output.vmfb
```

<details>
<summary>MLIR input (click to expand)</summary>

```mlir
module @module {
  
  func.func @main(%inv_variance_: !torch.tensor<[1,1,1,1],f32>, %mean_: !torch.tensor<[1,1,1,1],f32>, %y_: !torch.tensor<[1,128,64,32],f32>, %x: !torch.vtensor<[1,128,64,32],f32>) attributes {torch.assume_strict_symbolic_shapes} {
  
    %layernorm_fwd_EPSILON = torch.vtensor.literal(dense<0x3727C5AC> : tensor<1xf32>) : !torch.vtensor<[1],f32>

    %normalized_shape_val_0_layernorm_fwd = torch.constant.int 128
    %normalized_shape_val_1_layernorm_fwd = torch.constant.int 64
    %normalized_shape_val_2_layernorm_fwd = torch.constant.int 32
    %normalized_shape_layernorm_fwd = torch.prim.ListConstruct %normalized_shape_val_0_layernorm_fwd, %normalized_shape_val_1_layernorm_fwd, %normalized_shape_val_2_layernorm_fwd : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>

    
    %eps_layernorm_fwd = torch.aten.item %layernorm_fwd_EPSILON : !torch.vtensor<[1],f32> -> !torch.float

    %permute_x_val_0_layernorm_fwd = torch.constant.int 0
    %permute_x_val_1_layernorm_fwd = torch.constant.int 1
    %permute_x_val_2_layernorm_fwd = torch.constant.int 2
    %permute_x_val_3_layernorm_fwd = torch.constant.int 3
    %permute_x_layernorm_fwd = torch.prim.ListConstruct %permute_x_val_0_layernorm_fwd, %permute_x_val_1_layernorm_fwd, %permute_x_val_2_layernorm_fwd, %permute_x_val_3_layernorm_fwd : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>

    %x_layernorm_fwd_perm = torch.aten.permute %x, %permute_x_layernorm_fwd : !torch.vtensor<[1,128,64,32],f32>, !torch.list<int> -> !torch.vtensor<[1,128,64,32],f32>
  
    %none_scale_layernorm_fwd = torch.constant.none
    %none_bias_layernorm_fwd = torch.constant.none
    %y_layernorm_fwd_perm, %mean_layernorm_fwd_perm, %inv_variance_layernorm_fwd_perm = torch.aten.native_layer_norm %x_layernorm_fwd_perm, %normalized_shape_layernorm_fwd, %none_scale_layernorm_fwd, %none_bias_layernorm_fwd, %eps_layernorm_fwd : !torch.vtensor<[1,128,64,32],f32>, !torch.list<int>, !torch.none, !torch.none, !torch.float -> !torch.vtensor<[1,128,64,32],f32>, !torch.vtensor<[1,1,1,1],f32>, !torch.vtensor<[1,1,1,1],f32>
    %permute_y_val_0_layernorm_fwd = torch.constant.int 0
    %permute_y_val_1_layernorm_fwd = torch.constant.int 1
    %permute_y_val_2_layernorm_fwd = torch.constant.int 2
    %permute_y_val_3_layernorm_fwd = torch.constant.int 3
    %permute_y_layernorm_fwd = torch.prim.ListConstruct %permute_y_val_0_layernorm_fwd, %permute_y_val_1_layernorm_fwd, %permute_y_val_2_layernorm_fwd, %permute_y_val_3_layernorm_fwd : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>

    %y = torch.aten.permute %y_layernorm_fwd_perm, %permute_y_layernorm_fwd : !torch.vtensor<[1,128,64,32],f32>, !torch.list<int> -> !torch.vtensor<[1,128,64,32],f32>
  
    %permute_mean_val_0_layernorm_fwd = torch.constant.int 0
    %permute_mean_val_1_layernorm_fwd = torch.constant.int 1
    %permute_mean_val_2_layernorm_fwd = torch.constant.int 2
    %permute_mean_val_3_layernorm_fwd = torch.constant.int 3
    %permute_mean_layernorm_fwd = torch.prim.ListConstruct %permute_mean_val_0_layernorm_fwd, %permute_mean_val_1_layernorm_fwd, %permute_mean_val_2_layernorm_fwd, %permute_mean_val_3_layernorm_fwd : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>

    %mean = torch.aten.permute %mean_layernorm_fwd_perm, %permute_mean_layernorm_fwd : !torch.vtensor<[1,1,1,1],f32>, !torch.list<int> -> !torch.vtensor<[1,1,1,1],f32>
  
    %permute_inv_variance_val_0_layernorm_fwd = torch.constant.int 0
    %permute_inv_variance_val_1_layernorm_fwd = torch.constant.int 1
    %permute_inv_variance_val_2_layernorm_fwd = torch.constant.int 2
    %permute_inv_variance_val_3_layernorm_fwd = torch.constant.int 3
    %permute_inv_variance_layernorm_fwd = torch.prim.ListConstruct %permute_inv_variance_val_0_layernorm_fwd, %permute_inv_variance_val_1_layernorm_fwd, %permute_inv_variance_val_2_layernorm_fwd, %permute_inv_variance_val_3_layernorm_fwd : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>

    %inv_variance = torch.aten.permute %inv_variance_layernorm_fwd_perm, %permute_inv_variance_layernorm_fwd : !torch.vtensor<[1,1,1,1],f32>, !torch.list<int> -> !torch.vtensor<[1,1,1,1],f32>
  
    
    torch.overwrite.tensor.contents %inv_variance overwrites %inv_variance_ : !torch.vtensor<[1,1,1,1],f32>, !torch.tensor<[1,1,1,1],f32>
    torch.overwrite.tensor.contents %mean overwrites %mean_ : !torch.vtensor<[1,1,1,1],f32>, !torch.tensor<[1,1,1,1],f32>
    torch.overwrite.tensor.contents %y overwrites %y_ : !torch.vtensor<[1,128,64,32],f32>, !torch.tensor<[1,128,64,32],f32>

    return
  }
}
```
</details>

## Error

```
error: 'func.func' op failed to distribute
error: failed to run translation of source executable to target executable for backend #hal.executable.target<"rocm", "rocm-hsaco-fb", {abi = "hip", ...}>
```

## Observations

- **Fails**: input shape `[1,128,64,32]` (single batch)
- **Succeeds**: input shape `[16,128,64,32]` (multi batch)
- The normalized shape is `[128,64,32]` (all dims except batch), with no scale/bias (elementwise_affine=false)

## IREE version

- `iree-base-compiler` version: **3.11.0rc20260304**
- Wheel: `iree_base_compiler-3.11.0rc20260304-cp312-abi3-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl`
- IREE commit: [`a1ef7f98c3337d272f736c22276dcbdd4a1f1f55`](https://github.com/iree-org/iree/commit/a1ef7f98c3337d272f736c22276dcbdd4a1f1f55)
- Target: `gfx942` (MI300X / MI325X)

## CI reference

Fusilli CI failure: https://github.com/iree-org/fusilli/actions/runs/22681418514/job/65752673211?pr=220
